### PR TITLE
[Ide] Fix two deadlocks in Project class.

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs [hunk 1]
```diff
@@ -463,7 +463,7 @@ public Task<ProjectFile[]> GetSourceFilesAsync (ConfigurationSelector configurat
 
 				if (string.IsNullOrEmpty (coreCompileDependsOn)) {
 					evaluatedCompileItemsTask.SetResult (new ProjectFile [0]);
-					return evaluatedCompileItemsTask.Task.Result;
+					return await evaluatedCompileItemsTask.Task;
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs]: Is this really a deadlock?

Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs]: I don't think it is, I can get rid of that.

Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs]: Just saw the SetResult above.

Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs]: Just wondering if I missed something. Also, the first change is not a deadlock, although I'm ok with the change.

**Comment by Therzok** [comment]
Manually pushed to master 1a6dd5dc0652190afeda1684dd3a6d7a090c4043

**Pull request closed without merging**